restful_api_with_data_mapper/main.bal

```diff
@@ -0,0 +1,95 @@
+import ballerina/http;
+
+type Person record {
+    readonly string id;
+    string firstName;
+    string lastName;
+    int age;
+    string country;
+};
+
+type Course record {
+    string id;
+    string name;
+    int credits;
+};
+
+type Student record {
+    string id;
+    string fullName;
+    string age;
+    record {
+        string title;
+        int credits;
+    }[] courses;
+    int totalCredits;
+    string visaType;
+};
+
+const D_TIER_4_VISA = "D tier-4";
+
+var totalCredits = function(int total, record {string id; string name; int credits;} course) returns int => total + (course.id.startsWith("CS") ? course.credits : 0);
+
+function transform(Person person, Course[] courses) returns Student => let var isForeign = person.country != "LK" in {
```

Comment:
Shall we use meaningful name than `transform` as the function name? Even datamapper UI allows to define custome name IINM.

Reply:
Fixed via [d947cac](https://github.com/ballerina-guides/integration-samples/pull/27/commits/d947cacd24449500d642f90a284c9e224c1d244c)